packages/jitm/src/class-pre-connection-jitm.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Jetpack's Pre-Connection JITM class.
+ *
+ * @package automattic/jetpack-jitm
+ */
+
+namespace Automattic\Jetpack\JITMS;
+
+use Automattic\Jetpack\JITMS\JITM;
+
+/**
+ * Jetpack pre-connection just in time messaging through out the admin.
+ */
+class Pre_Connection_JITM extends JITM {
+
+	/**
+	 * Returns all the pre-connection messages.
+	 */
+	private function get_raw_messages() {
+		return array(
+			array(
+				'id'             => 'jpsetup-posts',
+				'message_path'   => '/wp:edit-post:admin_notices/',
+				'message'        => __( 'Do you know which of these posts gets the most traffic?', 'jetpack' ),
+				'description'    => __( 'Setup Jetpack to get in-depth stats about your content and visitors.', 'jetpack' ),
+				'button_link'    => esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-posts' ) ),
+				'button_caption' => __( 'Setup Jetpack', 'jetpack' ),
+			),
+			array(
+				'id'             => 'jpsetup-upload',
+				'message_path'   => '/wp:upload:admin_notices/',
+				'message'        => __( 'Do you want lightning-fast images?', 'jetpack' ),
+				'description'    => __( 'Setup Jetpack, enable Site Accelerator, and start serving your images lightning fast, for free.', 'jetpack' ),
+				'button_link'    => esc_url( \Jetpack::init()->build_connect_url( true, false, 'pre-connection-jitm-upload' ) ),
+				'button_caption' => __( 'Setup Jetpack', 'jetpack' ),
+			),
+			array(
+				'id'             => 'jpsetup-widgets',
+				'message_path'   => '/wp:widgets:admin_notices/',
+				'message'        => __( 'Looking for even more widgets?', 'jetpack' ),
+				'description'    => __( 'Setup Jetpack for great additional widgets like business hours and maps.', 'jetpack' ),
```

Comment:
Could we update the wording on this message, since we have no business Hours or Maps widget?

Reply:
According to @jessefriedman, we can keep it because the Contact Info widget does have hours https://jetpack.com/support/extra-sidebar-widgets/contact-info-widget/
So what should we do?

Reply:
@jeherve, I think we should keep it. The business contact widget includes a map and hours.

Reply:
Marking this one as Resolved. Feel free to open a new one.

Reply:
This still feels weird to me, especially since the wording mentions widget**s**, but if that's okay to you  ¯\_(ツ)_/¯